### PR TITLE
Tools fixes - s3cat, s3perf, http_speed

### LINKED ISSUE #7315: Put of a small object(1KB) by `s3cat --size` does not stop
### Environment info
- NooBaa Version: **master merged from https://github.com/guymguym/noobaa-core/tree/guy-p6 (believe it was caddda5e7512b0c7124bb9ebdaa66b5a1a4e6c56)**
- Platform: **stand-alone**

### Actual behavior
1. `node src/tools/s3cat` with `--size 0.001` (1K) option does not seem to end, and will keep showing "Upload Speed" for 20+ min.

    Waited for around 20 minutes before `ctl+c`-ing when `node src/tools/s3cat --endpoint <endpoint_url> --sig s3 --bucket <bucketname> --put <put_name> --size 0.001` was called:
    
    ```
    Uploading 1K-test from generated data of size 1.0 KB
    upload done. 0 MB/sec
    Upload Speed: 0.0 MB/sec (average 0.0)
    Upload Speed: 0.0 MB/sec (average 0.0)
    Upload Speed: 0.0 MB/sec (average 0.0)
    Upload Speed: 0.0 MB/sec (average 0.0)
    ....
    ....
    Upload Speed: 0.0 MB/sec (average 0.0)
    Upload Speed: 0.0 MB/sec (average 0.0)
    Upload Speed: 0.0 MB/sec (average 0.0)
    ^C
    ```

### Expected behavior
1. Command ends when `s3cat --put` with a small (or 0.001) `--size` is specified.

### Steps to reproduce
1. Run `s3cat --put` with `--size 0.001` option.

### More information - Screenshots / Logs / Other output
The `s3cat` worked and ended as expected when a 1k size file was specified with the `--file` option.
Maybe has something to do with having decimal vals in`data_size`?
https://github.com/noobaa/noobaa-core/blob/cc973768bbcb8d3b6e752eee0c64ce14af91d068/src/tools/s3cat.js#L267

### PR DESCRIPTION
**Pull request by guymguym**
### Explain the changes
1. fix s3cat put size fraction need rounding
2. s3perf improve get object bench
3. http_speed and tcp_speed use buffer pool
4. cleanup tools output by removing console.log from top level module code
5. fix nsfs and other cli tools that do not need db connection

### Issues: Fixed #xxx / Gap #xxx
1. Fixes #7315

### Testing Instructions:
1. NA

### CODE REVIEW

File: src/tools/http_speed.js [hunk 1]
```diff
@@ -40,6 +42,17 @@ const http_agent = argv.ssl ?
     new https.Agent({ keepAlive: true }) :
     new http.Agent({ keepAlive: true });
 
+const buffers_pool_sem = new Semaphore(1024*1024*1024, {
```
Comment (utkarsh-pro) [inline src/tools/http_speed.js:45]: I think linter is getting upset here.
```suggestion
const buffers_pool_sem = new Semaphore(1024 * 1024 * 1024, {
```

File: src/tools/s3perf.js [hunk 2]
```diff
@@ -191,25 +209,21 @@ async function run_worker_loop() {
...
-let _object_keys_next = 0;
-let _object_keys_done = false;
-let _object_keys_promise = null;
+/** @type {AWS.S3.ListObjectsOutput} */
+let _list_objects = { Contents: [], IsTruncated: true };
+let _list_objects_next = 0;
+let _list_objects_promise = null;
 
 /**
  * This function returns the next key to be used for head/get/delete.
```
Comment (utkarsh-pro) [inline src/tools/s3perf.js:229]:
```suggestion
 * This function returns the next object to be used for head/get/delete.
```

Comment (utkarsh-pro) [inline src/tools/s3perf.js:229]:
```suggestion
 * This function returns the next object to be used for head/get/delete.
```

File: src/tools/s3perf.js [hunk 3]
```diff
@@ -191,114 +209,124 @@ async function run_worker_loop() {
...
             .on('error', reject)
             .on('data', data => {
-                // noop
+                send_message({ ops: 0, size: data.length, took_ms: 0 });
             });
     });
+    return 0;
 }
 
 async function delete_all_objects() {
```
Comment (liranmauda) [inline src/tools/s3perf.js:301]: why is this `delete_all_objects`? seems like it deletes a single one...

**Review by liranmauda — APPROVED** [review]
LGTM, 
2 points: 
1. Seems like one function name (which was there and did not change) is not correct
2. s3pref is not supporting versioning. I guess it has no meaning in supporting that, just wanted to point it out.

**Pull request merged**